students/Jean-Baptiste/lessons/lesson02/assignment2_solution/Assignment02_solution_JB/charges_calc.py

```diff
@@ -0,0 +1,73 @@
+'''
+Returns total price paid for individual rentals
+'''
+import argparse
+import json
+import datetime
+import math
+import logging
+import traceback, sys
+
+
+log_format = "%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s %(message)s"
+#BEGIN NEW STUFF
+formatter = logging.Formatter(log_format)
+file_handler = logging.FileHandler('mylog.log')
+file_handler.setFormatter(formatter)
+logger = logging.getLogger()
+logger.addHandler(file_handler)
+ #END NEW STUFF
+
+def parse_cmd_arguments():
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument('-o', '--output', help='ouput JSON file', required=True)
+    return parser.parse_args()
+
+def load_rentals_file(filename):
+    with open(filename) as file:
+        try:
+            data = json.load(file)
+        except:
+            logging.error("There is an error loading json from file {}".format(filename))
+            logging.exception("message")
+            exit(1)
+    return data
+def calculate_additional_fields(data):
+    for value in data.values():
+        try:
+            rental_start = datetime.datetime.strptime(value['rental_start'], '%m/%d/%y')
+            rental_end = datetime.datetime.strptime(value['rental_end'], '%m/%d/%y')
+            value['total_days'] = (rental_end - rental_start).days
+            value['total_price'] = value['total_days'] * value['price_per_day']
+            value['sqrt_total_price'] = math.sqrt(value['total_price'])
+            if value['units_rented'] == 0:
+                value['unit_cost'] = value['total_price']
+            else:
+                value['unit_cost'] = value['total_price'] / value['units_rented']
+        except:
+            logging.error("There is a calcul error input data, please try one more time")
+            logging.exception("message")
+            exit(1)
+            value['unit_cost'] = value['total_price'] / value['units_rented']
+    return data
+
+''' To get utf8-encoded file as opposed to ascii-encoded,
+   json.dump writes to a text file, it's output is ASCII-only by default,
+'''
+def save_to_json(filename, data):
+    with open(filename, 'w',encoding='utf-8') as file:
+        json.dump(data, file, indent=4, ensure_ascii=False)
+def parse(self, inpath, outpath):
+    g = open(inpath, 'r')
+    with open(outpath, 'w') as fout:
+        for l in g:
+            fout.write(json.dumps(eval(l)))
+            clean_json=save_to_json
+            clean_json.parse("source.json", "cleaned.json")
+            clean_json.save()
+if __name__ == "__main__":
+    args = parse_cmd_arguments()
+    data = load_rentals_file(args.input)
+    data = calculate_additional_fields(data)
+    save_to_json(args.output, data)
```

Comment:
Not sure what you did in this file. You were suppose to add the ability to log from the command line, but I don't see that here.